### ISSUE TITLE
QGIS Desktop should hide QGIS Server only plugins

### ISSUE THREAD

**Issue by m-kuhn** [qgis/QGIS#36284]
It should be possible to hide qgis server only plugins from the qgis desktop plugin manager.

Current status (according to my understanding):

 - plugins can be tagged in metadata.txt with `server: True`
   I don't know if this setting is ever inspected by code somewhere and would make a difference.
 - plugins for server can be uploaded to the plugin repository.
   This is nice for visibility. I don't know if there is any mechanism in place or planned to install them from there on a server.

It would be nice to add a new metadata.txt variable `desktop: [boolean]` (True by default) which is used to filter plugins if qgis desktop loads the plugin list from the repository. Or list these plugins as not instalable on desktop to increase visibility but lower the risk of accidental installation.

Ideally it should be possible to connect the repository to qgis server too.

See https://github.com/opengisch/qgis_server_render_geojson/issues/3

Opinions?

CC @Gustry @elpaso

**Comment by elpaso** [comment]
Hi, AFAIK there are no plans for server direct installation of plugins, but a cli scripts to manage plugins would probably be nice to have.

As for the `server = True` tag, it is only used for filtering here: https://plugins.qgis.org/plugins/server/

The original idea was that a single plugin could have both server and desktop iface, I don't know if that makes still any sense. The use case at the time was a customer of mine's (believe it or not it was public administration of a provice) that was using a self-hosted debian machine as a QGIS Server + Desktop workstation.

**Comment by gioman** [comment]
> that was using a self-hosted debian machine as a QGIS Server + Desktop workstation.

seems to me that this is a pretty commom situation for testing/development purposes (and apparently also production).

**Comment by m-kuhn** [comment]
If we have a file that allows for both configs individually we can support both. With backwards compatibility for plugins that don't specify either.

```ini
server=True # Default to False
desktop=True # Default to True
```

**Comment by m-kuhn** [comment]
@elpaso I guess we need a change for that on the plugin repo infrastructure?

**Comment by elpaso** [comment]
yes, and in plugin manager too, it parses metadata.

**Comment by m-kuhn** [comment]
I am confident we find enough people to take care of the plugin manager.
Who has the knowledge to do that on server or at least provide pointers to code and deployment?

**Comment by elpaso** [comment]
I don't know about the deployment (@rduivenvoorde or @timlinux should know more), but the last time I worked on the plugin application I created a Vagrant recipe for local development: https://github.com/qgis/QGIS-Django/tree/modernize you will need to add a bool field for the new flag and get it from metadata parser, then add the field to the XML.

**Comment by m-kuhn** [comment]
ah cool, thanks! what's the state of this branch, it seems to have never been merged, does this matter?

**Comment by elpaso** [comment]
Yes, it does :/ . The modernize branch was the result of a paid work I did about one year ago where I updated to python3 and the latest Django version + I prepared the Vagrant to make it easier for others to maintain and develop the application, deployment was explicitly **not** part of the deal. I'm not sure about what happened next, but I think that it was never deployed. 

Now, there are probably some minor changes (I remember a small bugfix in the xml) that should be forward ported from master to modernize.

The change required for the desktop flag is really trivial, let me know if I can help. Perhaps it would be also good to have a validator rule to check for both flags being false.

**Comment by Gustry** [comment]
> Ideally it should be possible to connect the repository to qgis server too.

FYI, we have released a CLI tool to manage QGIS Server plugins. (even if it works as well for QGIS Desktop, it's designed for server).
https://github.com/3liz/qgis-plugin-manager
